1-js/02-first-steps/12-nullish-coalescing-operator/article.md

```diff
@@ -97,73 +97,73 @@ alert(height || 100); // 100
 alert(height ?? 100); // 0
 ```
 
-- The `height || 100` checks `height` for being a falsy value, and it's `0`, falsy indeed.
-    - so the result of `||` is the second argument, `100`.
-- The `height ?? 100` checks `height` for being `null/undefined`, and it's not,
-    - so the result is `height` "as is", that is `0`.
+- `height || 100` prüft `height` darauf, ob es ein falsy Wert ist, und das ist `0`, eindeutig falsy.
+    - daher ist das Ergebnis von `||` das zweite Argument (`100`).
+- `height ?? 100` prüft `height` darauf, ob es `null/undefined` ist, und das ist es nicht,
+    - daher ist das Ergebnis `height` "wie es ist", also `0`.
 
-In practice, the zero height is often a valid value, that shouldn't be replaced with the default. So `??` does just the right thing.
+In der Praxis ist die Höhe von Null oft ein gültiger Wert, der nicht durch den Standardwert ersetzt werden sollte. Daher macht `??` genau das Richtige.
 
-## Precedence
+## Vorrang
 
-The precedence of the `??` operator is the same as `||`. They both equal `3` in the [MDN table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table).
+Die Vorrangigkeit des `??` Operators ist die gleiche wie bei `||`. Beide sind gleich `3` in der [MDN-Tabelle](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table).
 
-That means that, just like `||`, the nullish coalescing operator `??` is evaluated before `=` and `?`, but after most other operations, such as `+`, `*`.
+Das bedeutet, dass genau wie `||` der Nullish Coalescing Operator `??` vor `=` und `?` ausgewertet wird, aber nach den meisten anderen Operationen, wie `+`, `*`.
 
-So we may need to add parentheses in expressions like this:
+Also könnten wir gezwungen sein, Klammern in Ausdrücken wie diesem hinzuzufügen:
 
 ```js run
 let height = null;
 let width = null;
 
-// important: use parentheses
+// wichtig: Klammern verwenden
 let area = (height ?? 100) * (width ?? 50);
 
 alert(area); // 5000
 ```
 
-Otherwise, if we omit parentheses, then as `*` has the higher precedence than `??`, it would execute first, leading to incorrect results.
+Wenn wir die Klammern weglassen, würde `*` als höhere Priorität als `??` zuerst ausgeführt, was zu falschen Ergebnissen führen würde.
 
 ```js
-// without parentheses
+// ohne Klammern
 let area = height ?? 100 * width ?? 50;
 
-// ...works this way (not what we want):
+// ...funktioniert so (nicht was wir wollen):
 let area = height ?? (100 * width) ?? 50;
 ```
 
-### Using ?? with && or ||
+### Verwendung von ?? mit && oder ||
 
-Due to safety reasons, JavaScript forbids using `??` together with `&&` and `||` operators, unless the precedence is explicitly specified with parentheses.
+Aus Sicherheitsgründen verbietet JavaScript die Verwendung von `??` zusammen mit `&&` und `||`, es sei denn die Vorrangregeln werden durch Klammern explizit angegeben.
```

Comment:
Die Übersetzung passt, aber der Satz ist von der Bedeutung her nicht ganz genau. Im Englishen ausgedrückt ist dieses Beispiel eine *Ambiguous Parsing Pathway* der in der Informatik unlösbar ist und daher zu einem Fehler führt und kommt vor wenn zwei operatoren die selbe Vorrangigkeit haben. Normalerweise ist das durch Definition gelöst sowie *In diesem Fall wird von links-nach-rechts ausgewertet*  aber führt oft zu unklarem Quellcode.